### DISCUSSION TITLE
Einen Befehl "Get-SampleCode" gibt es in den Listings nicht

### DISCUSSION THREAD

**Discussion by helmutkilian** [TobiasPSP/OReilly#7] [Allgemeines]
Buch S. 63, 65, 66, 74: 
Den dort erwähnten Befehl "Get-SampleCode" habe ich in den Listings und sonstigen Code-Zeilen nicht gefunden, sondern nur den gleichwertigen Befehl "L".

**Comment by TobiasPSP** [comment]
Das ist tatsächlich ein Druckfehler. Der Name wurde später in "L" geändert. Das entspricht zwar nicht guter Praxis für "normale" Befehle (Cmdlets), aber in diesem speziellen Fall sollte es so kurz wie möglich sein.

Der Quellcode des Befehls befindet sich hier: https://github.com/TobiasPSP/OReilly/blob/main/PowerShell/loadScript.ps1
Daraus wird dann mit dem folgenden Skript der neue Befehl "L":
https://github.com/TobiasPSP/OReilly/blob/main/PowerShell/Kapitel%201/1-15.ps1

Kurz gesagt liest das Skript den Quellcode direkt von github und legt ihn dann als Funktion namens "L" an. Man hätte die Funktion auch mit einem qualifizierteren Namen wie "Get-SampleCode" anlegen und danach einen Alias namens "L" auf diese Funktion anlegen können. Das wäre der übliche Weg bei "normalen" Befehlen. Da aber "L" nur interaktiv während der Lektüre des Buches genutzt wird und nicht in "normalen" Skripten als Befehlserweiterung, wäre dieser Weg ein "Overkill" ohne zusätzlichen Nutzen gewesen. 

Wobei das Meinungssache ist. Ein Vorteil wäre, dass die Funktion "L" auch unter dem "systematischeren" Namen "Get-SampleCode" wiedergefunden werden könnte, falls man sie vergessen hat.
